src/components/filters/filter_icons_overlay/filter_icons_overlay.ts

```diff
@@ -1,28 +1,29 @@
 import { Component } from "@odoo/owl";
-import { CellPosition, DOMCoordinates, SpreadsheetChildEnv } from "../../../types";
+import { CellPosition, SpreadsheetChildEnv } from "../../../types";
 import { GridCellIcon } from "../../grid_cell_icon/grid_cell_icon";
 import { FilterIcon } from "../filter_icon/filter_icon";
 
 interface Props {
-  gridPosition: DOMCoordinates;
+  onMouseDown: (ev: MouseEvent, closePopover: boolean) => void;
 }
 
 export class FilterIconsOverlay extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-FilterIconsOverlay";
   static props = {
-    gridPosition: { type: Object, optional: true },
+    onMouseDown: Function,
   };
   static components = {
     GridCellIcon,
     FilterIcon,
   };
-  static defaultProps = {
-    gridPosition: { x: 0, y: 0 },
-  };
 
   getFilterHeadersPositions(): CellPosition[] {
     const sheetId = this.env.model.getters.getActiveSheetId();
     const headerPositions = this.env.model.getters.getFilterHeaders(sheetId);
     return headerPositions.map((position) => ({ sheetId, ...position }));
   }
+
+  onMouseDown(ev: MouseEvent) {
+    this.props.onMouseDown(ev, false); // don't close popover on icon click
```

Comment:
Not a big fan of this. This is needed otherwise the mouse down would close the filter menu (grid onCellClicked), then the menu will be re-open on the onClick of the filter Icon.